flake.nix

```diff
@@ -1,61 +1,36 @@
 {
   description = "nix.dev static website";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.05";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
   inputs.flake-utils.url = "github:numtide/flake-utils/master";
-  inputs.poetry2nix = {
-    inputs.flake-utils.follows = "flake-utils";
-    inputs.nixpkgs.follows = "nixpkgs";
-    url = "github:nix-community/poetry2nix/master";
-  };
 
-  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
+  outputs = { self, nixpkgs, flake-utils}:
     flake-utils.lib.eachDefaultSystem (system:
       let
         pkgs = import nixpkgs {
           inherit system;
-          overlays = [ poetry2nix.overlay ];
         };
-        poetryOverrides = self: super: {
-
-          # Refs https://github.com/nix-community/poetry2nix/issues/218#issuecomment-981615612
-          typing-extensions = super.typing-extensions.overridePythonAttrs (
-            old: {
-              buildInputs = (old.buildInputs or [ ]) ++ [ self.flit-core ];
-            }
-          );
-
-          sphinx-design = super.sphinx-design.overridePythonAttrs (
-            old: {
-              buildInputs = (old.buildInputs or [ ]) ++ [ self.flit-core ];
-            }
-          );
+      in {
+        defaultPackage = pkgs.stdenv.mkDerivation {
+          name = "nix-dev";
+          src = self;
+          buildInputs = with pkgs.python310Packages; [
+            livereload
```

Comment:
`livereload` is only used for local development as it is a tool which invokes a build command every time a change is detected.

Reply:
Are you suggesting I do something differently?

Reply:
@zmitchell Yeah, I'm just saying it's not necessary to be inside `buildInputs` since it is not a dependency required for building the website.

Reply:
Ok, that part I get. What I don't get is where `black` and `livereload` are supposed to go instead. The `live` script relies on `livereload` being available for the call to `nix-shell --run`, and since I'm new to Nix I'm not sure the best way to achieve the same thing if those Python modules aren't in `buildInputs`.

Reply:
A good approach is to have the main derivation describing only the tools and libraries it needs to build, and then have a separate derivation corresponding to a development shell where you add additional tools.